### ISSUE TITLE
Add Crockford's Base32 support to @std/encoding OR Add the ability to use a custom dictionary

### ISSUE THREAD

**Issue by wesleyakio** [denoland/std#6235]
**Is your feature request related to a problem? Please describe.**

I currently in some projects use UUID columns in PostgreSQL to store ULIDs in a storage optimized format. I use @std/ulid to generate those ULIDs. However converting them to binary and then hex is done by copying @std/encoding/_base32_common.ts into my project and using it with the dictionary from @std/ulid/_util.ts. Considering the project already uses both packages, I'd appreciate an alternative solution to copying the files into my project.

**Describe the solution you'd like**

There are many solutions that cover my use case:
1) Adding a base32Crockford.ts just like we have base32.ts and unstable_base32hex.ts
2) Exposing the encode/decode methods from @std/encoding/_base32_common.ts
3) Since base32 and base32hex already seem to share 100% of the code and only differ on the dictionary, perhaps simply going from
```typescript
export function decodeBase32(b32: string): Uint8Array {
  return decode(b32, lookup);
}
```
to
```typescript
export function decodeBase32(b32: string, options: Base32Options = { lookup }): Uint8Array {
  return decode(b32, options.lookup);
}
```
Would do the trick for all 3 cases.

If going with option 3, adding constant export for the dictionaries such as:
```typescript
export const BASE32_DICT: string[] = "ABCDEFGHIJKLMNOPQRSTUVWXYZ234567".split("");
export const BASE32HEX_DICT:  string[] = "0123456789ABCDEFGHIJKLMNOPQRSTUV".split("");
export const CROCKFORD_DICT:  string[] = "0123456789ABCDEFGHJKMNPQRSTVWXYZ".split("");
```
would also be very helpful.

**Describe alternatives you've considered**

Right now I have @std/encoding code copied into my project. I could use other libraries by I'm trying to keep to @std as much as I can.

**Comment by wesleyakio** [comment]
BTW, I'm happy to submit a PR if one of the solutions above would be acceptable/wanted.

**Comment by kt3k** [comment]
I think solution 1 would be most aligned to the current direction of that package. (creating `unstable_base32_crockford.ts` and exposing it as `@std/encoding/unstable-base32-crockford`)

BTW I'm curious why you choose crockford base32 over other format.

**Comment by wesleyakio** [comment]
I opened a PR https://github.com/denoland/std/pull/6238 for this, ended up going with `unstable_base32crockford.ts` instead of `unstable_base32_crockford.ts` in order to mimic the already existing `unstable_base32hex.ts`, can change if needed.

**Comment by timreichen** [comment]
Out of interest: why is it `32hex` and not `32_hex`?

**Comment by kt3k** [comment]
> but we have ULIDs in @std/ulid and do not have v7s on @std/uuid so this seemed like the lightest lift.

I'm not sure you are aware or not, but UUID v7 is available as unstable API at `@std/uuid/unstable-v7`. https://jsr.io/@std/uuid/doc/unstable-v7

When this is available, does that work better than the combination of ULID (binary) + crockford base32?

**Comment by kt3k** [comment]
fixed by #6238
